app/admin.go

```diff
@@ -168,32 +169,19 @@ func validateLdapFilter(cfg *model.Config, ldap einterfaces.LdapInterface) *mode
 	return ldap.ValidateFilter(*cfg.LdapSettings.UserFilter)
 }
 
-func (a *App) SaveConfig(cfg *model.Config, sendConfigChangeClusterMessage bool) *model.AppError {
-	oldCfg := a.Config()
-	cfg.SetDefaults()
-	a.Desanitize(cfg)
-
-	if err := cfg.IsValid(); err != nil {
-		return err
-	}
-
-	if err := validateLdapFilter(cfg, a.Ldap); err != nil {
+func (a *App) SaveConfig(newCfg *model.Config, sendConfigChangeClusterMessage bool) *model.AppError {
+	if err := validateLdapFilter(newCfg, a.Ldap); err != nil {
```

Comment:
Note that `validateLdapFilter` now occurs before the `IsValid` check inside `configStore.Set`. As you note above, `validateLdapFilter` only requires the `Enable` and `UserFilter` from the incoming config, so validity here wasn't an issue. For the time being, I opted not to embed this check into the config store itself in the hope that I could think about it as an "external" validity vs. an "internal" one. 

I'm not 100% sold on this -- because the enterprise code will `panic` if it discovers the filter to be invalid, so it may as well be "internal". There's a lot about the config subsystem that we could stand to improve. Thoughts welcome here.

Reply:
This comment is out of date: the `validateLdapFilter` now occurs as part of the model validity checks.